packages/shared/src/components/modals/streaks/StreakRecoverModal.tsx

```diff
@@ -0,0 +1,229 @@
+        fill="#FC538D"
+      />
+    </svg>
+  </div>
+);
+
+const StreakRecoverHeading = ({ days }: { days: number }) => (
+  <Typography
+    className="-mx-1 -mb-1 text-center font-bold"
+    tag={TypographyTag.H3}
+    type={TypographyType.Title1}
+    data-testid="streak-recover-modal-heading"
+  >
+    Oh no! Your
+    <span className="text-accent-bacon-default"> {days} day streak </span> has
+    been broken!
+  </Typography>
+);
+
+const StreakRecoveryCopy = ({
+  reputation,
+  cost,
+}: Record<'reputation' | 'cost', number>) => {
+  const isFree = cost === 0;
+  const canRecover = reputation >= cost;
+  const reputationLink = (
+    <a
+      target="_blank"
+      rel={anchorDefaultRel}
+      href={reputationHref}
+      title="What are reputation points?"
+      className="text-text-link underline-offset-2 hover:underline"
+    >
+      reputation points
+    </a>
+  );
+
+  const isFreeText = (
+    <>
+      Lucky you! The first streak restore is on us 🎁. This usually costs 25{' '}
+      {reputationLink}. Be sure to come prepared next time!
+    </>
+  );
+  const canRecoverText = (
+    <>
+      Maintain your streak!
+      <br />
+      Use {cost} {reputationLink} to keep going.
+    </>
+  );
+  const noRecoverText = (
+    <>You don’t have enough {reputationLink} to restore your streaks.</>
+  );
+
+  return (
+    <Typography
+      className="text-center"
+      tag={TypographyTag.P}
+      type={TypographyType.Body}
+      data-testid="streak-recovery-copy"
+    >
+      {isFree && isFreeText}
+      {!isFree && (canRecover ? canRecoverText : noRecoverText)}
+    </Typography>
+  );
+};
+
+const StreakRecoverButton = ({
+  cost,
+  ...props
+}: Record<'cost', number> & ButtonProps<'button'>) => (
+  <Button
+    {...props}
+    className="relative"
+    variant={ButtonVariant.Primary}
+    size={ButtonSize.Large}
+    data-testid="streak-recover-button"
+  >
+    Restore my streak
+    <span className="absolute -right-0.5 -top-0.5 flex items-center whitespace-nowrap rounded-bl-14 rounded-tr-14 bg-accent-onion-default px-1.5 py-1 text-text-primary">
```

Comment:
Why do we need all this?

Reply:
Yes, this is the style for this badge 👀 
<img width="260" alt="image" src="https://github.com/user-attachments/assets/0307e343-4f17-446b-b2d3-ddc68bd9c827">